lib/libc/newlib/libc-hooks.c

```diff
@@ -557,7 +557,7 @@ void *_sbrk_r(struct _reent *r, int count)
 
 int _gettimeofday(struct timeval *__tp, void *__tzp)
 {
-#ifdef CONFIG_POSIX_API
+#if defined(CONFIG_POSIX_API) || defined(CONFIG_POSIX_CLOCK)
```

Comment:
I would say the real bug is that `POSIX_CLOCK` does not depend on `POSIX_API`.

@cfriedt what do you think?

Reply:
I don't think the `POSIX_CLOCK` should require full POSIX API.

At least on nRF SDK, I seem to end up in various conflicts with overloaded socket API, if full POSIX is turned on. But the POSIX CLOCK is provided and therefore works OK.

Reply:
That works